src/main/java/com/hms_networks/sc/cumulocity/CConnectorMain.java

```diff
@@ -1,24 +1,620 @@
+      Logger.LOG_EXCEPTION(e);
+      initializeSuccess = false;
+    }
+
+    if (initializeSuccess) {
+      Logger.LOG_CRITICAL("Finished initializing " + CONNECTOR_FRIENDLY_NAME + ".");
+    } else {
+      Logger.LOG_CRITICAL("Failed to initialize " + CONNECTOR_FRIENDLY_NAME + "!");
+    }
+    return initializeSuccess;
+  }
+
+  /** Method for performing connector application initialization steps. */
+  private static boolean startUp() {
+    Logger.LOG_CRITICAL("Starting " + CONNECTOR_FRIENDLY_NAME + "...");
+    boolean startUpSuccess = true;
+
+    setUpConnectorControlTag();
+
+    // Run MQTT provisioning manager if provisioning required
+    try {
+      if (!connectorConfig.isProvisioned()) {
+        startUpSuccess &= runProvisioningMqtt();
+      }
+    } catch (Exception e) {
+      Logger.LOG_CRITICAL("Unable to check existing provisioning status of connector.");
+      Logger.LOG_EXCEPTION(e);
+    }
+
+    // Start main MQTT manager
+    startUpSuccess &= setUpMqtt();
+
+    // Create tag alarm manager
+    try {
+      alarmMgr = new CConnectorAlarmMgr();
+      Logger.LOG_CRITICAL("Created the tag alarm manager.");
+    } catch (Exception e) {
+      Logger.LOG_CRITICAL("Failed to create the tag alarm manager!");
+      Logger.LOG_EXCEPTION(e);
+    }
+
+    if (startUpSuccess) {
+      Logger.LOG_CRITICAL("Finished starting " + CONNECTOR_FRIENDLY_NAME + ".");
+    } else {
+      Logger.LOG_CRITICAL("Failed to start " + CONNECTOR_FRIENDLY_NAME + "!");
+    }
+    return startUpSuccess;
+  }
+
+  /** Method for performing connector main loop tasks. */
+  private static void runMainLoop() {
+    Logger.LOG_DEBUG("Running main loop...");
+
+    // Grab latest data and send via MQTT
+    try {
+      CConnectorDataMgr.checkForHistoricalDataAndSend(mqttMgr);
+    } catch (Exception e) {
+      Logger.LOG_CRITICAL("Unable to grab latest data and send via MQTT.");
+      Logger.LOG_EXCEPTION(e);
+    }
+
+    // Update connector control tag value
+    if (connectorControlTag != null && isRunning) {
+      isRunning = (connectorControlTag.getTagValueAsInt() == CONNECTOR_CONTROL_TAG_RUN_VALUE);
+    }
+  }
+
+  /** Method for performing connector application initialization steps. */
+  private static void shutDown() {
+    Logger.LOG_CRITICAL("Shutting down " + CONNECTOR_FRIENDLY_NAME + "...");
+    boolean shutDownClean = true;
+
+    // Disconnect from MQTT
+    if (mqttMgr != null) {
+      try {
+        mqttMgr.disconnect();
+      } catch (Exception e) {
+        Logger.LOG_CRITICAL("Unable to disconnect from MQTT.");
+        shutDownClean = false;
+      }
+    }
+
+    if (shutDownClean) {
+      Logger.LOG_CRITICAL("Finished shutting down " + CONNECTOR_FRIENDLY_NAME + ".");
+    } else {
+      Logger.LOG_CRITICAL("Failed to shut down " + CONNECTOR_FRIENDLY_NAME + " properly!");
+    }
+  }
+
+  /** Method for performing connector application initialization steps. */
```

Comment:
update comment

Reply:
addressed this for startUp(), cleanUp() and shutDown()